src/components/MessageBox/MessageBox.spec.js

```diff
@@ -1,11 +1,18 @@
 import '@testing-library/jest-dom';
 import {render, screen} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
 
 import MessageBox from './index';
 
-describe('MessageBox component', () => {
+const mockedUsedNavigate = jest.fn();
+jest.mock('react-router-dom', () => ({
+	...jest.requireActual('react-router-dom'),
+	useNavigate: () => mockedUsedNavigate,
+}));
```

Comment:
I don't think you actually need this. When testing a component which uses a feature from react-router, like `useNavigate` it should be sufficient enough to wrap your component with the `MemoryRouter` from react-router. Here is a link to the docs about the testing library used with react router: https://testing-library.com/docs/example-react-router/

However: it is PRETTY cool, that you managed to write your own mock function! 🥳 And if it works, it is not per se wrong 🙂. Just want to give an alternative.

Reply:
I will look at that.